packages/fxa-auth-server/lib/account-delete.ts

```diff
@@ -242,15 +242,65 @@ export class AccountDeleteManager {
     });
   }
 
+  public async refundSubscriptions(
+    uid: string,
+    deleteReason: ReasonForDeletion,
+    refundPeriod?: number
+  ) {
+    // Currently only support auto refund of invoices for unverified accounts
+    if (deleteReason !== 'fxa_unverified_account_delete') {
+      return;
+    }
+
+    const createdDate = refundPeriod
+      ? new Date(new Date().setDate(new Date().getDate() - refundPeriod))
+      : undefined;
+    const invoices =
+      await this.stripeHelper?.fetchInvoicesForActiveSubscriptions(
+        uid,
+        'paid',
+        createdDate
+      );
+
+    if (!invoices?.length) {
+      // Log message and return
+      return;
+    }
+
+    // Attempt Stripe and PayPal refunds
+    const refundResults = await Promise.all([
+      this.stripeHelper?.refundInvoices(invoices),
+      this.paypalHelper?.refundInvoices(invoices),
+    ]);
+    refundResults.forEach((results) =>
+      results?.forEach((result) =>
+        this.log.info('AccountDeleteManager.refundSubscriptions', {
+          invoiceId: result.invoiceId,
+          priceId: result.priceId,
+          total: result.total,
+          currency: result.currency,
+        })
```

Comment:
Why not log as they're refunded to ensure any errors don't stop the logging of the refunds that did happen?

Reply:
Mostly just to make it easier to pick up the logs. This way it's all being logged in one place, instead of 2 different methods.

Reply:
I'll move it back into the individual methods. Should make the code simpler as well.